features/algebraic-type-templated-matching.feature

```diff
@@ -31,9 +31,10 @@ Feature: Outputting Algebraic Types With Templated Matching
       #import "SimpleADT.h"
       #import <memory>
 
-      namespace SimpleADTMatching {
-        template <typename T>
-        extern T match(SimpleADT *simpleADT, T(^firstSubtypeMatchHandler)(NSString *firstValue, NSUInteger secondValue), T(^someRandomSubtypeMatchHandler)(), T(^someAttributeSubtypeMatchHandler)(NSUInteger someAttributeSubtype), T(^secondSubtypeMatchHandler)(BOOL something)) {
+      template <typename T>
+      struct SimpleADTMatching {
```

Comment:
A gerund is not ideal for a struct (since in objective-c land, it usually means a protocol). I'd call it:

   *Matcher (in this case, SimpleADTMatcher)